### ISSUE TITLE
Technical community discussion questions

### ISSUE THREAD

**Issue by Qiqutimes** [Sugaritem/Technical-community#1]
1、以太坊区块链中的每个数据都经过哈希处理。那是默克尔树的形式。因此如何检索数据成为可能，因为一旦对数据进行哈希处理就不可能颠倒了吗？
1. Every data in ETH blockchain is hashed. That is the form of the Merkel tree. So how to retrieve data becomes possible, because once the data is hashed, it is impossible to reverse it?

**Comment by 0xguy** [comment]
That's correct, is impossible to reverse and decrypt hashed data on the ETH network, unless during encryption that hash and the data has been leaked. 

However, there are some tool on the internet that contains a mapping of various hashed data with original data, but that chances to find result for your own hashed data are very low.

**Comment by 0xlemon** [comment]
# 重塑树

想要重塑默克尔树，就需要按照创建时完全相同的数据和顺序，更改某一处数据或循序，都将改变默克尔树。

# 校验检索

对于两份复杂数据校验，寻找差异部分，只需要将两份数据的默克尔树进行比较。
从根Hash依次向下进行比较，只需要检索Hash不同的部分，就能定位异常数据块，从而快速找到恶意数据。

------

# Reshape the tree

If you want to reshape the Merkel tree, you need to change the data or order in a certain place according to the exact same data and sequence when it was created, which will change the entire Merkel tree.

# Verify search

For the verification of two complex data, to find the difference, only need to compare the Merkel trees of the two data.
Compare from the root Hash to the bottom. Only by retrieving the different parts of the Hash, the abnormal data block can be located and the malicious data can be found quickly.
